### PR TITLE
change(Balsamic): Remove concatenation

### PR DESCRIPTION
**Pull request by ivadym**
### Changed
- Removed Balsamic concatenation. Will be done in Balsamic and submitted as a slurm job.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py [hunk 1]
```diff
@@ -234,21 +249,54 @@ def test_error_single_wgs_panel_arg(
...
+        in caplog.text
+    )
 
 
-def test_error_two_tumor(balsamic_context: CGConfig, cli_runner: CliRunner, caplog):
-    """Test with case_id that requires WGS analysis but has TWO tumor ONE normal samples"""
+def test_error_two_tumor(
+    balsamic_context: CGConfig, cli_runner: CliRunner, caplog: LogCaptureFixture
+):
+    """Test with case_id that requires WGS analysis but has TWO tumor ONE normal samples."""
```
Comment (henrikstranneheim) [inline tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py:299]:
```suggestion
    """Test with case_id that requires WGS analysis but has TWO tumor, and ONE normal samples."""
```

File: tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py [hunk 2]
```diff
@@ -304,20 +360,24 @@ def test_error_mixed_panel_bed_resque(balsamic_context: CGConfig, cli_runner: Cl
...
-    assert "Invalid number of normal samples" in caplog.text
+    assert f"Case {case_id} has an invalid number of samples" in caplog.text
 
 
-def test_error_wes_panel(balsamic_context: CGConfig, cli_runner: CliRunner, caplog):
-    """Test with case_id containing ONE tumor and TWO normal samples"""
+def test_error_wes_panel(
+    balsamic_context: CGConfig, cli_runner: CliRunner, caplog: LogCaptureFixture
+):
+    """Test with case_id containing ONE tumor and TWO normal samples."""
```
Comment (henrikstranneheim) [inline tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py:380]: Seems incorrect and needs an update

File: tests/meta/workflow/conftest.py [hunk 3]
```diff
@@ -195,3 +200,33 @@ def qc_microsalt_context(
...
+@pytest.fixture(name="balsamic_sample_data")
+def fixture_balsamic_sample_data(
+    sample_id: str,
+    cust_sample_id: str,
+    another_sample_id: str,
+    bed_file: str,
+) -> Dict[str, dict]:
+    """Balsamic sample data dictionary."""
+    return {
+        sample_id: {
```
Comment (henrikstranneheim) [inline tests/meta/workflow/conftest.py:214]: No hard coding - Awesome! 👍

**Review by henrikstranneheim — COMMENTED** [review]
Looking good so far 👍

File: tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py (hunk 2 above)
Comment (ivadym) [reply tests/cli/workflow/balsamic/test_cli_balsamic_config_case.py:380]: Good catch 👍

File: cg/meta/upload/scout/balsamic_config_builder.py [hunk 4]
```diff
@@ -32,36 +32,36 @@ def include_case_files(self):
...
         sample_id: str = config_sample.sample_id
         if config_sample.alignment_path:
-            if "tumor" in config_sample.alignment_path:
-                sample_id = "tumor"
-            elif "normal" in config_sample.alignment_path:
-                sample_id = "normal"
+            if SampleType.TUMOR in config_sample.alignment_path:
+                sample_id = SampleType.TUMOR.value
+            elif SampleType.NORMAL in config_sample.alignment_path:
+                sample_id = SampleType.NORMAL.value
```
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/meta/upload/scout/balsamic_config_builder.py:42]: Nice, in this way we can also have some pytests ⭐

File: cg/meta/workflow/balsamic.py [hunk 5]
```diff
@@ -613,15 +559,15 @@ def family_has_correct_number_tumor_normal_samples(self, case_id: str) -> bool:
         """Evaluates if a case has exactly one tumor and up to one normal sample in ClinicalDB.
         This check is only applied to filter jobs which start automatically"""
         query = (
```
[the remark this answers is not in the document]
Comment (ivadym) [reply cg/meta/workflow/balsamic.py:561]: Actually these methods are not used anywhere, so I will just get rid of them (`family_has_correct_number_tumor_normal_samples` and `get_valid_cases_to_analyze`)

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/meta/workflow/balsamic.py]: Great! Even better

**Comment by ivadym** [comment]
Closing as it has diverged from Master and current implementation.

**Pull request closed without merging**